components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,376 @@
+
+@Composable
+fun CalculateBorder(
+    enabled: Boolean = true,
+    isError: Boolean,
+    interactionSource: InteractionSource,
+    focusedBorderThickness: Dp = FocusedBorderThickness,
+    unfocusedBorderThickness: Dp = UnfocusedBorderThickness
+) {
+    borderStroke = animateBorderStrokeAsState(
+        enabled,
+        isError,
+        interactionSource,
+        focusedBorderThickness,
+        unfocusedBorderThickness
+    )
+}
+
+@Composable
+fun indicatorColor(
+    enabled: Boolean,
+    isError: Boolean,
+    interactionSource: InteractionSource
+): State<Color> {
+    val focused by interactionSource.collectIsFocusedAsState()
+
+    val colors = HmrcTheme.colors
+
+    val focusedBorderColor = colors.hmrcBlue
+    val errorBorderColor = colors.hmrcBlack
+    val unfocusedBorderColor = colors.hmrcBlack
+
+    val targetValue = when {
+        isError -> errorBorderColor
+        focused -> focusedBorderColor
+        else -> unfocusedBorderColor
+    }
+    return if (enabled) {
+        animateColorAsState(targetValue, tween(durationMillis = ANIMATION_DURATION))
+    } else {
+        rememberUpdatedState(targetValue)
+    }
+}
+
+@SuppressWarnings("LongMethod", "ComplexMethod")
+@Composable
+fun PasswordTextInputView(
+    modifier: Modifier = Modifier,
+    value: String? = null,
+    onInputValueChange: ((String) -> Unit)? = null,
+    inputFilter: ((String, String) -> String)? = null,
+    labelText: String? = null,
+    labelContentDescription: String? = null,
+    hintText: String? = null,
+    hintContentDescription: String? = null,
+    prefix: @Composable() (() -> Unit)? = null,
+    placeholderText: String? = null,
+    errorText: String? = null,
+    errorContentDescription: String? = null,
+    showButtonText: String? = stringResource(R.string.passwordTextInputView_show),
+    hideButtonText: String? = stringResource(R.string.passwordTextInputView_hide),
+    showButtonContentDescription: String? = null,
+    hideButtonContentDescription: String? = null,
+    showButtonStateDescription: String? = null,
+    hideButtonStateDescription: String? = null,
+    characterCount: Int? = null,
+    maxChars: Int? = null,
+    singleLine: Boolean = false,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    interactionSource: MutableInteractionSource,
+    requiredSequencesSpacing: Boolean,
+    textStyle: TextStyle
+) {
+    var hideButtonShown: Boolean by rememberSaveable { mutableStateOf(false) }
+
+    var localValue: String by rememberSaveable { mutableStateOf(value.orEmpty()) }
+    localValue = value.orEmpty()
+
+    var localError: String? by rememberSaveable { mutableStateOf(null) }
+    localError = errorText
+
+    var isError: Boolean by rememberSaveable { mutableStateOf(false) }
+
+    isError = !localError.isNullOrEmpty() || (localValue.length > (characterCount ?: Int.MAX_VALUE))
+
+    @Composable
+    fun ShowHideButton(
+        modifier: Modifier,
```

Comment:
We don't need this modifier parameter as we never pass anything through here to it.